pkg/p2p/libp2p/parser.go

```diff
@@ -0,0 +1,53 @@
+package libp2p
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	"github.com/ethersphere/bee/pkg/bzz"
+	"github.com/ethersphere/bee/pkg/crypto"
+	"github.com/ethersphere/bee/pkg/p2p/libp2p/internal/handshake/pb"
+	"github.com/ethersphere/bee/pkg/swarm"
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+type MockParser struct {
+	networkID uint64
+}
+
+var ErrInvalidAddress = errors.New("invalid address")
+
+func generateSignData(underlay, overlay []byte, networkID uint64) []byte {
+	networkIDBytes := make([]byte, 8)
+	binary.BigEndian.PutUint64(networkIDBytes, networkID)
+	signData := append([]byte("bee-handshake-"), underlay...)
+	signData = append(signData, overlay...)
+	return append(signData, networkIDBytes...)
+}
+
+func (m *MockParser) ParseAck(ack *pb.Ack, _ []byte) (*bzz.Address, error) {
```

Comment:
not sure why this is needed? can't we somehow tell libp2p to initialize the handshake service in a way that just never verifies the overlay? or in a sense the handshake service would just plug a different noop `parseAck` function that would always approve the ack?